go-controller/pkg/ovn/egressip.go

```diff
@@ -760,6 +770,82 @@ func findReroutePolicyIDs(filterOption, egressIPName string, gatewayRouterIP net
 	return strings.Split(policyIDs, "\n"), nil
 }
 
+func (oc *Controller) checkEgressNodesReachability() {
+	for {
+		reAddOrDelete := map[string]bool{}
+		oc.eIPC.allocatorMutex.Lock()
+		for _, eNode := range oc.eIPC.allocator {
+			if eNode.isEgressAssignable {
+				wasOnline := eNode.isOnline
+				isOnline := oc.isReachable(eNode)
+				if wasOnline && !isOnline {
+					reAddOrDelete[eNode.name] = true
+				} else if !wasOnline && isOnline {
+					reAddOrDelete[eNode.name] = false
+				}
+				eNode.isOnline = isOnline
```

Comment:
So this seems to be mixing the "node ready" and "node reachable" states... I think we want to keep track of them separately so that a node only gets egress IPs if it is both `Ready` and reachable.

Reply:
Isn't the reason that it's not done like that for openshift-sdn: that it takes a "long time" for the node object to be flagged as `NotReady`? Whereas we would like to detect this as fast as possible and re-assign all egress IPs away from that node?

Reply:
Never mind, I get what you mean and we can do both. Which means: we can track both states independently and only use a node for egress assignment once both are true, yet have the egress IP re-assignment be done when either are false.